[RHIDP-11676]: Add log aggregation and OpenTelemetry documentation (#2082)
redhat-developer/red-hat-developers-documentation-rhdh: modules/extend_orchestrator-in-rhdh/ref-process-lifecycle-events-for-timeline-tracking.adoc

:_mod-docs-content-type: REFERENCE

[id="process-lifecycle-events-for-timeline-tracking_{context}"]
= Process lifecycle events for timeline tracking

[role="_abstract"]
OpenTelemetry events that mark workflow execution milestones including start, completion, errors, and state transitions. Use these events to reconstruct the chronological order of workflow execution and identify when failures occurred.

To understand the exact chronological order of a workflow's execution, look for OpenTelemetry events that are automatically generated at key lifecycle points. SonataFlow attaches the following events and their specific attributes to spans to mark execution milestones:

> * `process.instance.start`: Marks the beginning of the execution. It includes the `process.instance.id`, what `trigger` started it, and the `reference.id`.

Reviewer: ```suggestion
* `process.instance.start`: Indicates the beginning of the execution. This event includes the `process.instance.id`, the `trigger` that started the process, and the `reference.id`.
```

Reviewer: Added these suggestions for parellelism.

> * `process.instance.complete`: Marks the end of the workflow. It includes the `process.instance.id`, the final `outcome`, and the total `duration.ms`.

Reviewer: ```suggestion
* `process.instance.complete`: Indicates the completion of the workflow. This event includes the `process.instance.id`, the final `outcome`, and the total `duration.ms`.
```

> * `process.instance.error`: Added when a workflow fails. It includes the `process.instance.id`, the `error.message`, and the `error.type`.

Reviewer: ```suggestion
* `process.instance.error`: Indicates a workflow failure. This event includes the `process.instance.id`, the `error.message`, and the `error.type`.
```

> * `state.started` and `state.completed`: Generated for each individual workflow state. Both include an `event.description` detailing the state execution.

Reviewer: ```suggestion
* `state.started` and `state.completed`: Indicate the start and completion of individual workflow states. These events include an `event.description` that details the state execution.
```

> * `log.message`: Includes the actual application log content within the trace span. It provides the `level`, `logger`, `message`, `thread.name`, and `thread.id`.

Reviewer: ```suggestion
* `log.message`: Indicates the application log content within the trace span. This event provides the `level`, `logger`, `message`, `thread.name`, and `thread.id`.
```
